src/Symfony/Component/Security/Core/Authentication/Token/AbstractToken.php

```diff
@@ -38,32 +36,16 @@ abstract class AbstractToken implements TokenInterface
     public function __construct(array $roles = [])
     {
         foreach ($roles as $role) {
-            if (\is_string($role)) {
-                $role = new Role($role, false);
-            } elseif (!$role instanceof Role) {
-                throw new \InvalidArgumentException(sprintf('$roles must be an array of strings, or Role instances, but got %s.', \gettype($role)));
-            }
-
-            $this->roles[] = $role;
-            $this->roleNames[] = (string) $role;
+            $this->roleNames[] = $role;
```

Comment:
The old code made sure that `$role` is a string. Shouldn't we keep the check?

Reply:
I don't think so. We only needed the cast before as `$role` was a `Role` instance then. Now we expect an array of strings.

Reply:
what @derrabus means is the `InvalidArgumentException` being thrown above in case of wrong type.

Reply:
Yes, but we usually don't do that anywhere else but expect the user to respect the types specified in the docblock, right?